payment/controllers.go

```diff
@@ -0,0 +1,353 @@
+			return handlers.WrapError(err, "Error encoding the transaction JSON", http.StatusInternalServerError)
+		}
+		return nil
+	})
+}
+
+// CreateAnonCardTransactionRequest includes information needed to create a anon card transaction
+type CreateAnonCardTransactionRequest struct {
+	WalletID    uuid.UUID `json:"paymentId"`
+	Transaction string    `json:"transaction"`
+}
+
+// CreateAnonCardTransaction creates a transaction against an order
+func CreateAnonCardTransaction(service *Service) handlers.AppHandler {
+	return handlers.AppHandler(func(w http.ResponseWriter, r *http.Request) *handlers.AppError {
+		var req CreateAnonCardTransactionRequest
+		err := requestutils.ReadJSON(r.Body, &req)
+		if err != nil {
+			return handlers.WrapError(err, "Error in request body", http.StatusBadRequest)
+		}
+
+		orderID := chi.URLParam(r, "orderID")
+		if orderID == "" || !govalidator.IsUUIDv4(orderID) {
+			return &handlers.AppError{
+				Message: "Error validating request url parameter",
+				Code:    http.StatusBadRequest,
+				Data: map[string]interface{}{
+					"orderID": "orderID must be a uuidv4",
+				},
+			}
+		}
+		validOrderID := uuid.Must(uuid.FromString(orderID))
+
+		transaction, err := service.CreateAnonCardTransaction(r.Context(), req.WalletID, req.Transaction, validOrderID)
+		if err != nil {
+			return handlers.WrapError(err, "Error creating anon card transaction", http.StatusInternalServerError)
+		}
+
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusCreated)
+		if err := json.NewEncoder(w).Encode(transaction); err != nil {
+			return handlers.WrapError(err, "Error encoding the transaction JSON", http.StatusInternalServerError)
+		}
+		return nil
+
+	})
+}
+
+// CreateOrderCredsRequest includes the item ID and blinded credentials which to be signed
+type CreateOrderCredsRequest struct {
+	ItemID       uuid.UUID `json:"itemId" valid:"-"`
+	BlindedCreds []string  `json:"blindedCreds" valid:"base64"`
+}
+
+// CreateOrderCreds is the handler for creating order credentials
+func CreateOrderCreds(service *Service) handlers.AppHandler {
+	return handlers.AppHandler(func(w http.ResponseWriter, r *http.Request) *handlers.AppError {
+		var req CreateOrderCredsRequest
+		err := requestutils.ReadJSON(r.Body, &req)
+		if err != nil {
+			return handlers.WrapError(err, "Error in request body", http.StatusBadRequest)
+		}
+
+		_, err = govalidator.ValidateStruct(req)
+		if err != nil {
+			return handlers.WrapValidationError(err)
+		}
+
+		orderID := chi.URLParam(r, "orderID")
+		if orderID == "" || !govalidator.IsUUIDv4(orderID) {
+			return &handlers.AppError{
+				Message: "Error validating request url parameter",
+				Code:    http.StatusBadRequest,
+				Data: map[string]interface{}{
+					"orderID": "orderID must be a uuidv4",
+				},
+			}
+		}
+		validOrderID := uuid.Must(uuid.FromString(orderID))
+
+		err = service.CreateOrderCreds(r.Context(), validOrderID, req.ItemID, req.BlindedCreds)
```

Comment:
we should double check, I think this is another potential idempotency issue - we may need to modify `CreateOrderCreds` to first see if they were already successfully saved

Reply:
Sounds good! did something similar where I added a http status 409 after calling `GetOrderCreds`
https://github.com/brave-intl/bat-go/pull/226/commits/c7f78c25cd9f92a6488ab6da92eeb5a8f3c5c7e1